youtube_dl/extractor/platzi.py

```diff
@@ -104,50 +104,40 @@ def _real_extract(self, url):
         lecture_id = self._match_id(url)
 
         webpage = self._download_webpage(url, lecture_id)
-
-        data = self._parse_json(
-            self._search_regex(
-                # client_data may contain "};" so that we have to try more
-                # strict regex first
-                (r'client_data\s*=\s*({.+?})\s*;\s*\n',
-                 r'client_data\s*=\s*({.+?})\s*;'),
-                webpage, 'client data'),
+        data_preloaded_state = self._parse_json(
+            self._search_regex((r'window.__PRELOADED_STATE__ = (.*)\<\/script'), webpage, 'client data'),
```

Comment:
You could make the regex more robust:
```
r'window\s*.\s*__PRELOADED_STATE__\s*=\s*(.*?)\s*</script'
```
* `\s*` for all the places where JS/HTML allow whitespace
* `.*?` to avoid capturing "target_json</script>...<script...>...</script>", etc
* `</` don't need to be escaped.

Also, as you're presumably hoping to get some brace expression, maybe this?
```
r'window\s*.\s*__PRELOADED_STATE__\s*=\s*({.*?});?\s*</script'
```